opcua/server/address_space.py

```diff
@@ -180,34 +180,46 @@ def add_nodes(self, addnodeitems, user=User.Admin):
     def _add_node(self, item, user):
         result = ua.AddNodesResult()
 
-        if item.RequestedNewNodeId in self._aspace:
+        # Check if the null NodeId is given
+        if item.RequestedNewNodeId.is_null():
+            self.logger.debug("RequestedNewNodeId is null")
+            if item.BrowseName.NamespaceIndex:
```

Comment:
OK, but I would like just to remark something...

This will imply bigger changes since in the test it is written that you create NodeId with the NamespaceIndex 4, but since the parent is `Objects` folder you will add it into NamespaceIndex 0 (which is by the way forbidden by the standard). So if we always use the NamespaceIndex of the parent there is no way to have the Nodes in other NamespaceIndex than 0, and you can not specify other NamespaceIndex for the child node (which should be possible)....

Maybe I understood you wrong, please correct me...